packages/mui-utils/src/useControlled/useControlled.test.js

```diff
@@ -105,4 +105,10 @@ describe('useControlled', () => {
       setProps({ defaultValue: 1 });
     }).not.toErrorDev();
   });
+
+  it('should not raise a warning if setting NaN as the defaultValue when uncontrolled', () => {
+    expect(() => {
+      render(<TestComponent defaultValue={NaN}>{() => null}</TestComponent>);
+    }).not.toErrorDev();
+  });
```

Comment:
Why should we support this?

Reply:
Bundle size shouldn’t be impacted at all because the whole effect is wrapped in a `process.env.NODE_ENV` check, and nowhere else in the codebase uses `Object.is` and there are `IE11` comments, so we figured it’s better to keep the [partial IE11 support](https://mui.com/material-ui/getting-started/supported-platforms/#ie-11)

Reply:
> One example is if you have a Select component where one of the options has value of `NaN`

Isn't it wrong that an option has a value of `NaN`?